tests/rptest/services/redpanda.py

```diff
@@ -1833,6 +1907,46 @@ def write_node_conf_file(self,
             doc["redpanda"].update(dict(kafka_api_tls=tls_config))
             conf = yaml.dump(doc)
 
+        if self._pandaproxy_config is not None and self._pandaproxy_config.pp_client_key is not None and self._pandaproxy_config.pp_client_crt is not None:
+            pp_tls_config = dict(
+                key_file=RedpandaService.TLS_SERVER_KEY_FILE,
+                cert_file=RedpandaService.TLS_SERVER_CRT_FILE,
+                truststore_file=RedpandaService.TLS_CA_CRT_FILE,
+                enabled=True,
+                require_client_auth=True)
+
+            client_tls_config = dict(
+                key_file=PandaproxyConfig.PP_TLS_CLIENT_KEY_FILE,
+                cert_file=PandaproxyConfig.PP_TLS_CLIENT_CRT_FILE,
+                truststore_file=RedpandaService.TLS_CA_CRT_FILE,
+                enabled=True,
+                require_client_auth=True)
+            doc = yaml.full_load(conf)
+            doc["pandaproxy"].update(dict(pandaproxy_api_tls=pp_tls_config))
+            doc["pandaproxy_client"].update(dict(broker_tls=client_tls_config))
+            conf = yaml.dump(doc)
+
+        if self._schema_registry_config is not None and self._schema_registry_config.sr_client_key is not None and self._schema_registry_config.sr_client_crt is not None:
+            sr_tls_config = dict(
+                key_file=RedpandaService.TLS_SERVER_KEY_FILE,
+                cert_file=RedpandaService.TLS_SERVER_CRT_FILE,
+                truststore_file=RedpandaService.TLS_CA_CRT_FILE,
+                enabled=True,
+                require_client_auth=True)
+
+            client_tls_config = dict(
+                key_file=SchemaRegistryConfig.SR_TLS_CLIENT_KEY_FILE,
+                cert_file=SchemaRegistryConfig.SR_TLS_CLIENT_CRT_FILE,
+                truststore_file=RedpandaService.TLS_CA_CRT_FILE,
+                enabled=True,
+                require_client_auth=True)
+            doc = yaml.full_load(conf)
+            doc["schema_registry"].update(
+                dict(schema_registry_api_tls=sr_tls_config))
+            doc["schema_registry_client"].update(
+                dict(broker_tls=client_tls_config))
+            conf = yaml.dump(doc)
```

Comment:
nit: I wonder whether this should be here or in the template.

Reply:
> nit: I wonder whether this should be here or in the template.

I may not understand your question but I don't see how we can put this in the yaml template directly. 

I can agree that wrapping this logic into a parent class would be more readable.

Reply:
```
{% if schema_registry_config is not none and ... %}
schema_registry_client:
  broker_tls:
    key_file: {{ schema_registry_config.SR_TLS_CLIENT_KEY_FILE }}
    cert_file: {{ schema_registry_config.SR_TLS_CLIENT_CRT_FILE }}
    truststore_file: {{ The CA probably needs injecting }}
    enabled: true
```

Reply:
OH that's what you mean. Yeah I can do that, I have not fully woken up yet.